server/src/main/java/org/cloudfoundry/identity/uaa/client/VmidentityQueryableClientDetailsService.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ *     Cloud Foundry
+ *     Copyright (c) [2017] VMware, Inc. All Rights Reserved.
+ *
+ *     This product is licensed to you under the Apache License, Version 2.0 (the "License").
+ *     You may not use this product except in compliance with the License.
+ *
+ *     This product includes a number of subcomponents with
+ *     separate copyright notices and license terms. Your use of these
+ *     subcomponents is subject to the terms and conditions of the
+ *     subcomponent's license, as noted in the LICENSE file.
+ *******************************************************************************/
+package org.cloudfoundry.identity.uaa.client;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cloudfoundry.identity.uaa.resources.Queryable;
+import org.cloudfoundry.identity.uaa.resources.QueryableResourceManager;
+import org.cloudfoundry.identity.uaa.zone.MultitenantVmidentityClientDetailsService;
+import org.springframework.security.oauth2.provider.ClientDetails;
+
+public class VmidentityQueryableClientDetailsService implements Queryable<ClientDetails>,
+                QueryableResourceManager<ClientDetails> {
+
+    private static final Log logger = LogFactory.getLog(VmidentityQueryableClientDetailsService.class);
+
+    private MultitenantVmidentityClientDetailsService delegate;
+
+    protected VmidentityQueryableClientDetailsService(MultitenantVmidentityClientDetailsService delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public List<ClientDetails> query(String filter, String sortBy, boolean ascending) {
+        logger.debug("query filtering not yet implemented");
```

Comment:
The query call is not yet implemented as we don't have much in the way of story for sending the generic query to IDM. I don't think this is strictly SCIM-related (in the JDBC version they just wrap the filter directly into a bigger SQL query), but as part of the SCIM work, we'll need to introduce a generic querying mechanism and can figure out what we want to do with the filters in this case.